Fix `generate-types.sh` and add an integration test to check that rtk generated endpoints are sync with `openapi.yml`

### What happened?

Changes in the back are not cached or tested this as lead to an error when I tried to generate rtk endpoints:
```
yarn run v1.22.19
$ exec scripts/generate-types.sh
Generating ../common/api/osrdMiddlewareApi.ts
{
  stack: 'ResolverError: Error opening file "/home/bloussou/osrd/api/openapi.yaml" \n' +
    "ENOENT: no such file or directory, open '/home/bloussou/osrd/api/openapi.yaml'\n" +
    '    at ReadFileContext.callback (/home/bloussou/osrd/front/node_modules/@apidevtools/json-schema-ref-parser/lib/resolvers/file.js:52:20)\n' +
    '    at FSReqCallback.readFileAfterOpen [as oncomplete] (node:fs:319:13)',
  code: 'ERESOLVER',
  message: 'Error opening file "/home/bloussou/osrd/api/openapi.yaml" \n' +
    "ENOENT: no such file or directory, open '/home/bloussou/osrd/api/openapi.yaml'",
  source: '/home/bloussou/osrd/api/openapi.yaml',
  path: null,
  toJSON: [Function: toJSON],
  ioErrorCode: 'ENOENT',
  name: 'ResolverError',
  toString: [Function: toString]
}
```

### What did you expect to happen?

To work at any time and that none of the generated `MiddlewareApi.ts` should eb updateedd if I haven't modified `openapi.yml` files

### How can we reproduce it (as minimally and precisely as possible)?

.

### What operating system, browser and environment are you using?

.


### OSRD version (top right corner `Account` button > `Informations`)

.

--- a/front/src/common/api/osrdEditoastApi.ts
+++ b/front/src/common/api/osrdEditoastApi.ts
@@ -1,5 +1,4 @@
 import { baseEditoastApi as api } from './emptyApi';
-
 const injectedRtkApi = api.injectEndpoints({
   endpoints: (build) => ({
     getHealth: build.query<GetHealthApiResponse, GetHealthApiArg>({
@@ -132,7 +131,7 @@ const injectedRtkApi = api.injectEndpoints({
         url: `/infra/${queryArg.id}/pathfinding/`,
         method: 'POST',
         body: queryArg.body,
-        params: { number: queryArg.number },
+        params: { number: queryArg['number'] },
       }),
     }),
     postInfraByIdObjectsAndObjectType: build.mutation<
@@ -154,6 +153,16 @@ const injectedRtkApi = api.injectEndpoints({
     >({
       query: () => ({ url: `/electrical_profile_set/` }),
     }),
+    postElectricalProfileSet: build.mutation<
+      PostElectricalProfileSetApiResponse,
+      PostElectricalProfileSetApiArg
+    >({
+      query: (queryArg) => ({
+        url: `/electrical_profile_set`,
+        method: 'POST',
+        params: { name: queryArg.name },
+      }),
+    }),
     getElectricalProfileSetById: build.query<
       GetElectricalProfileSetByIdApiResponse,
       GetElectricalProfileSetByIdApiArg
@@ -167,7 +176,19 @@ const injectedRtkApi = api.injectEndpoints({
       query: (queryArg) => ({ url: `/electrical_profile_set/${queryArg.id}/level_order/` }),
     }),
     getDocumentsByKey: build.query<GetDocumentsByKeyApiResponse, GetDocumentsByKeyApiArg>({
-      query: (queryArg) => ({ url: `/documents/${queryArg.key}` }),
+      query: (queryArg) => ({ url: `/documents/${queryArg.key}/` }),
+    }),
+    deleteDocumentsByKey: build.mutation<
+      DeleteDocumentsByKeyApiResponse,
+      DeleteDocumentsByKeyApiArg
+    >({
+      query: (queryArg) => ({ url: `/documents/${queryArg.key}/`, method: 'DELETE' }),
+    }),
+    postDocuments: build.mutation<PostDocumentsApiResponse, PostDocumentsApiArg>({
+      query: () => ({ url: `/documents/`, method: 'POST' }),
+    }),
+    getRollingStockById: build.query<GetRollingStockByIdApiResponse, GetRollingStockByIdApiArg>({
+      query: (queryArg) => ({ url: `/rolling_stock/${queryArg.id}/` }),
     }),
   }),
   overrideExisting: false,
@@ -415,6 +436,11 @@ export type GetElectricalProfileSetApiResponse =
     name?: string;
   }[];
 export type GetElectricalProfileSetApiArg = void;
+export type PostElectricalProfileSetApiResponse =
+  /** status 200 The list of ids and names of electrical profile sets available */ ElectricalProfile;
+export type PostElectricalProfileSetApiArg = {
+  name: string;
+};
 export type GetElectricalProfileSetByIdApiResponse =
   /** status 200 The list of electrical profiles in the set */ ElectricalProfile[];
 export type GetElectricalProfileSetByIdApiArg = {
@@ -431,8 +457,23 @@ export type GetElectricalProfileSetByIdLevelOrderApiArg = {
 };
 export type GetDocumentsByKeyApiResponse = unknown;
 export type GetDocumentsByKeyApiArg = {
-  /** A key identifying the document, often given by another endpoint */
+  /** A key identifying the document */
   key: number;
+};
+export type DeleteDocumentsByKeyApiResponse = unknown;
+export type DeleteDocumentsByKeyApiArg = {
+  /** A key identifying the document */
+  key: number;
+};
+export type PostDocumentsApiResponse = /** status 201 The key of the added document */ {
+  document_key?: number;
+};
+export type PostDocumentsApiArg = void;
+export type GetRollingStockByIdApiResponse =
+  /** status 200 The rolling stock information */ RollingStock;
+export type GetRollingStockByIdApiArg = {
+  /** Rolling Stock ID */
+  id: number;
 };
 export type SearchQuery = (boolean | number | number | string | SearchQuery)[] | null;
 export type ViewMetadata = {
@@ -561,4 +602,70 @@ export type ElectricalProfile = {
   value?: string;
   power_class?: string;
   track_ranges?: TrackRange[];
+};
+export type LightRollingStock = {
+  id?: number;
+  version?: string;
+  name?: string;
+  length?: number;
+  max_speed?: number;
+  startup_time?: number;
+  startup_acceleration?: number;
+  comfort_acceleration?: number;
+  gamma?: {
+    type?: 'CONST' | 'MAX';
+    value?: number;
+  };
+  inertia_coefficient?: number;
+  features?: string[];
+  mass?: number;
+  rolling_resistance?: {
+    A?: number;
+    B?: number;
+    C?: number;
+    type?: 'davis';
+  };
+  loading_gauge?: 'G1' | 'G2' | 'GA' | 'GB' | 'GB1' | 'GC' | 'FR3.3' | 'FR3.3/GB/G2' | 'GLOTT';
+  effort_curves?: {
+    default_mode?: string;
+    modes?: {
+      [key: string]: {
+        is_electric?: boolean;
+      };
+    };
+  };
+  base_power_class?: string;
+  power_restrictions?: {
+    [key: string]: string;
+  };
+  metadata?: object;
+  liveries?: {
+    id?: number;
+    name?: string;
+  }[];
+};
+export type Comfort = 'AC' | 'HEATING' | 'STANDARD';
+export type EffortCurve = {
+  speeds?: number[];
+  max_efforts?: number[];
+};
+export type ConditionalEffortCurve = {
+  cond?: {
+    comfort?: Comfort | null;
+    electrical_profile_level?: string | null;
+    power_restriction_code?: string | null;
+  } | null;
+  curve?: EffortCurve;
+};
+export type RollingStock = LightRollingStock & {
+  effort_curves?: {
+    default_mode?: string;
+    modes?: {
+      [key: string]: {
+        curves?: ConditionalEffortCurve[];
+        default_curve?: EffortCurve;
+        is_electric?: boolean;
+      };
+    };
+  };
 };

--- a/front/src/common/api/osrdMiddlewareApi.ts
+++ b/front/src/common/api/osrdMiddlewareApi.ts
@@ -862,6 +862,10 @@ export type LightRollingStock = {
       };
     };
   };
+  base_power_class?: string;
+  power_restrictions?: {
+    [key: string]: string;
+  };
   metadata?: object;
   liveries?: {
     id?: number;
@@ -875,7 +879,9 @@ export type EffortCurve = {
 };
 export type ConditionalEffortCurve = {
   cond?: {
-    comfort?: Comfort;
+    comfort?: Comfort | null;
+    electrical_profile_level?: string | null;
+    power_restriction_code?: string | null;
   } | null;
   curve?: EffortCurve;
 };
@@ -937,6 +943,11 @@ export type Allowance =
   | ({
       allowance_type: 'StandardAllowance';
     } & StandardAllowance);
+export type PowerRestrictionRange = {
+  begin_position?: number;
+  end_position?: number;
+  power_restriction_code?: string;
+};
 export type TrainScheduleOptions = {
   ignore_electrical_profiles?: boolean | null;
 };
@@ -952,6 +963,7 @@ export type StandaloneSimulationParameters = {
     allowances?: Allowance[];
     speed_limit_tags?: string;
     comfort?: Comfort;
+    power_restriction_ranges?: PowerRestrictionRange[] | null;
     options?: TrainScheduleOptions | null;
   }[];
 };

--- a/front/src/config/openapi-middleware-config.ts
+++ b/front/src/config/openapi-middleware-config.ts
@@ -1,7 +1,7 @@
 import type { ConfigFile } from '@rtk-query/codegen-openapi';
 
 const config: ConfigFile = {
-  schemaFile: '../../../api/openapi.yaml',
+  schemaFile: '../../../python/api/openapi.yaml',
   apiFile: '../common/api/emptyApi.ts',
   apiImport: 'baseApi',
   outputFile: '../common/api/osrdMiddlewareApi.ts',

--- a/tests/tests/test_rtk_sync.py
+++ b/tests/tests/test_rtk_sync.py
@@ -1,0 +1,17 @@
+import subprocess
+from pathlib import Path
+
+FRONT_FOLDER = Path(__file__).parent.parent.parent / "front"
+MIDDLEWARE_FOLDER = FRONT_FOLDER / "src" / "common" / "api"
+MIDDLEWARE_FILES = {
+    "api": MIDDLEWARE_FOLDER / "osrdEditoastApi.ts",
+    "editoast": MIDDLEWARE_FOLDER / "osrdMiddlewareApi.ts",
+}
+
+
+def test_rtk_middlewares_are_in_sync_with_openapi_files():
+    current_content = {name: file.read_text() for name, file in MIDDLEWARE_FILES.items()}
+    subprocess.check_output(["yarn", "generate-types"], cwd=FRONT_FOLDER)
+    updated_content = {name: file.read_text() for name, file in MIDDLEWARE_FILES.items()}
+    for name in MIDDLEWARE_FILES.keys():
+        assert current_content[name] == updated_content[name]

